components/server/src/authorization/authorizer.ts

```diff
@@ -418,17 +418,33 @@ export class Authorizer {
         if (await this.isDisabled(userID)) {
             return;
         }
-        const rels = [
-            set(rel.workspace(workspaceID).org.organization(orgID)),
-            set(rel.workspace(workspaceID).owner.user(userID)),
-        ];
-        if (shared) {
-            rels.push(set(rel.workspace(workspaceID).shared.anyUser));
-        }
+        return this.bulkAddWorkspaceToOrg([{ orgID, userID, workspaceID, shared }]);
+    }
 
+    async bulkAddWorkspaceToOrg(
+        ids: { orgID: string; userID: string; workspaceID: string; shared: boolean }[],
+    ): Promise<void> {
+        const rels: v1.RelationshipUpdate[] = [];
+        for (const { orgID, userID, workspaceID, shared } of ids) {
+            this.internalAddWorkspaceToOrg(orgID, userID, workspaceID, shared, (u) => rels.push(u));
+        }
         await this.authorizer.writeRelationships(...rels);
     }
 
+    private internalAddWorkspaceToOrg(
```

Comment:
reuse the same logic for single and bulk additions, as it was already out of sync